sentinel.go

```diff
@@ -0,0 +1,92 @@
+package sentinel
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+func New(ctx context.Context, config Config) *sentinel {
```

Comment:
I think it would be interesting to enable users to pass any channel they want and use that for the triggering mechanism. If the trigger channel ever closed Sentinel could detect this and treat as "Stop".